test/jdk/java/awt/Canvas/MultiDitherTest.java

```diff
@@ -0,0 +1,426 @@
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/*
+ * @test
+ * @bug 6616089
+ * @summary Displays a dithered Canvas on all available GraphicsConfigurations
+ * @library /java/awt/regtesthelpers
+ * @build PassFailJFrame
+ * @run main/manual MultiDitherTest
+ */
+
+import java.awt.BorderLayout;
+import java.awt.Button;
+import java.awt.Canvas;
+import java.awt.Choice;
+import java.awt.Color;
+import java.awt.Dimension;
+import java.awt.EventQueue;
+import java.awt.FlowLayout;
+import java.awt.FontMetrics;
+import java.awt.Frame;
+import java.awt.Graphics;
+import java.awt.GraphicsConfiguration;
+import java.awt.GraphicsDevice;
+import java.awt.GraphicsEnvironment;
+import java.awt.Image;
+import java.awt.Label;
+import java.awt.LayoutManager;
+import java.awt.Panel;
+import java.awt.TextField;
+import java.awt.event.ActionEvent;
+import java.awt.event.ActionListener;
+import java.awt.event.WindowAdapter;
+import java.awt.event.WindowEvent;
+import java.awt.image.ColorModel;
+import java.awt.image.MemoryImageSource;
+import java.util.List;
+
+public class MultiDitherTest extends Panel implements Runnable {
+    final static int NOOP = 0;
+    final static int RED = 1;
+    final static int GREEN = 2;
+    final static int BLUE = 3;
+    final static int ALPHA = 4;
+    final static int SATURATION = 5;
+    final static String calcString = "Calculating...";
+    static LayoutManager dcLayout = new FlowLayout(FlowLayout.CENTER, 10, 5);
+    Thread runner;
+    DitherControls XControls;
+    DitherControls YControls;
+    DitherCanvas canvas;
+
+    public static void main(String[] args) throws Exception {
+        String INSTRUCTIONS = """
+                This test displays several dithered Canvases similar to the
+                DitherTest demo applet, one for each available GraphicsConfiguration.
+                Depending on the GraphicsConfiguration, the dithering may be in
+                color or in grayscale and/or display at a lower bitdepth.
+                The number of GraphicsConfigurations will be printed in the
+                TextArea below as the test is starting up..
+                On Solaris, making some Windows active will cause some of the
+                other Windows to display strangely. This is normal.
+                Ensure that there are as many Frames created as there are
```

Comment:
remove the sentence about Solaris, it hasn't been a supported platform for several years

Reply:
Removed.